app/scripts/models/account.js

```diff
@@ -16,31 +16,41 @@ define([
   'lib/constants',
   'models/profile-image'
 ], function (Backbone, _, p, AuthErrors, ProfileClient, Constants, ProfileImage) {
-
   // Account attributes that can be persisted
   var PERSISTENT = {
-    uid: undefined,
+    accessToken: undefined,
     email: undefined,
+    grantedPermissions: undefined,
+    lastLogin: undefined,
+    needsOptedInToMarketingEmail: undefined,
+    profileImageId: undefined,
+    profileImageUrl: undefined,
     sessionToken: undefined,
     sessionTokenContext: undefined,
-    accessToken: undefined,
-    verified: undefined,
-    profileImageUrl: undefined,
-    profileImageId: undefined,
-    lastLogin: undefined,
-    grantedPermissions: undefined
+    uid: undefined,
+    verified: undefined
   };
 
   var DEFAULTS = _.extend({
-    password: undefined,
-    unwrapBKey: undefined,
+    customizeSync: undefined,
     keyFetchToken: undefined,
-    customizeSync: undefined
+    password: undefined,
+    unwrapBKey: undefined
   }, PERSISTENT);
 
   var ALLOWED_KEYS = Object.keys(DEFAULTS);
   var ALLOWED_PERSISTENT_KEYS = Object.keys(PERSISTENT);
 
+  function withMarketingEmailClient(method) {
+    return function () {
+      var self = this;
+      return self.getOAuthTokenFromSessionToken(self.get('sessionToken'), 'basket')
```

Comment:
> is this correct?

I realized I didn't answer this explicitly: yes, this looks correct, modulo potentially changing it to "basket:write".

An alternative approach would be to request a single accessToken with scope "profile:write basket:write" that we can persist to the account and use for both purposes.  I can't think of compelling reasons to favour one over the other though.